### PR TITLE
this commit will return an array of comments when given a specific article_id and handle errors

### PR DESCRIPTION
**Pull request by Hasan-M12**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -112,8 +112,45 @@ describe("GET /api/articles", () => {
...
+      .then(({ body }) => {
+        const allComments = body.comments;
+        allComments.forEach((eachComment) => {
+          expect(eachComment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number),
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:134]: The comments for a specific article have been requested so we should assert the correct id on each comments

File: __tests__/app.test.js [hunk 2]
```diff
@@ -112,8 +112,45 @@ describe("GET /api/articles", () => {
...
 });
+
+describe("GET /api/articles/articles_id/comments", () => {
+  test("200: should respond with an array of aall comment from article with a given article_id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const allComments = body.comments;
+        allComments.forEach((eachComment) => {
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:127]: As we are using assertions within a forEach we need to also assert in this test the correct length of the array. If the endpoint sends back an empty array (which is should not for this test) currently those asertions won't run and so the test will give a false positive. the length assertion will protect against this

File: __tests__/app.test.js [hunk 3]
```diff
@@ -112,8 +112,45 @@ describe("GET /api/articles", () => {
...
+  test("200: should return an array of comments in a descending order by created_at", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const allComments = body.comments;
+        expect(allComments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("GET 404: should give an appropriate error message when given an invalid endpoint", () => {
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:148]: This test will have passed straight away because of your `wrongPathHandler`. You only need to test for this once (which you have done) not for each task.

You do need to test for a 404 if the id is valid but nonexistent though for this endpoint eg `/api/articles/99999/comments`.

You also need a test for an invalid id too eg  `/api/articles/banana/comments`.

File: controllers/app.controllers.js [hunk 4]
```diff
@@ -40,3 +40,15 @@ exports.getArticles = (req, res, next) => {
       next(err);
     });
 };
+
+exports.getCommentsByArticlesId = (req, res, next) => {
+  const id = req.params;
+  const articleId = id.article_id
```
Comment (nc-mezz-davies) [inline controllers/app.controllers.js:46]: This is a little confusing. I would look into the notes on destructuring but I think the code you need is `const { articleId } = req.params` - this would give you the article_id variable to pass to selectArticlesByID

File: controllers/app.controllers.js [hunk 5]
```diff
@@ -40,3 +40,15 @@ exports.getArticles = (req, res, next) => {
...
+  return Promise.all([selectArticlesByID(articleId), selectCommentsByArticleId(articleId)])
```
Comment (nc-mezz-davies) [inline controllers/app.controllers.js:47]: There is no need to send the article back on the response. We do want to know if that article exists though as this will determine the reason for there being no comments see the comments on missing tests.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -112,8 +112,45 @@ describe("GET /api/articles", () => {
...
+    return request(app)
+      .get("/api/articles/1/banana")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Incorrect endpoint")
+      });
+  });
+});
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:156]: We also need a test that an existing article that has no comments returns the empty array. We need this to be the case, for example a brand new article - it would exist but not have any comments. This would NOT be an error.
